content/client-lib-development-guide/features.textile

```diff
@@ -55,6 +56,35 @@ h2(#test-guidelines). Test guidelines
 * @(G3)@ Testing statistics can be tricky due to timing issues and slow test suites as a result of sending requests to generate statistics.  As such, we provide a special stats endpoint in our sandbox environment that allows stats to be injected into our metrics system so that stats tests can make predictable assertions.  To create stats you must send an authenticated @POST@ request to the stats JSON to @https://sandbox-rest.ably.io/stats@ with the stats data you wish to create. See the "Javascript stats fixture":https://github.com/ably/ably-js/blob/4e65d4e13eb8750a375b9511e4dd059092c0e481/spec/rest/stats.test.js#L8-L51 and "setup helper":https://github.com/ably/ably-js/blob/4e65d4e13eb8750a375b9511e4dd059092c0e481/spec/common/modules/testapp_manager.js#L158-L182 as an example
 * @(G4)@ This spec defines API version 1.2. A client library must identify to Ably the version of the spec it uses in all requests and connections, per "RSC7a":#RSC7a and "RTN2f":#RTN2f. The spec it uses is defined as the latest API version for which the library implements all spec items relating to the wire protocol
 
+h2(#endpoint). Ably service endpoints
+
+* @(RAS1)@ Clients access the Ably service via multiple endpoints. A set of "@ClientOptions@":#options are supported that allow users to specify non-default values for the hostnames for those endpoints. REST operations are performed against a primary REST host and, for realtime client, connections are made against a primary realtime host.
+* @(RAS2)@ The primary REST host and primary realtime host are determined based on the supplied @ClientOptions@ as follows.
+** @(RAS2a)@ If the @ClientOptions#restHost@ option is set, the specified host is the primary REST host. It is not valid for @ClientOptions#restHost@ to be specified in conjunction with either a @ClientOptions#environment@ or @ClientOptions#zone@ option.
+** @(RAS2b)@ If the @ClientOptions#realHost@ option is set, the specified host is the primary realtime host. It is not valid for @ClientOptions#realHost@ to be specified in conjunction with either a @ClientOptions#environment@ or @ClientOptions#zone@ option.
+** @(RAS2b)@ Otherwise, the primary hosts are determined by first determining values for @zone@ and @environment@ as follows.
+*** @(RAS2b1)@ The value of @zone@ determines the principal domain (PD) of the primary hosts. If the @ClientOptions#zone@ option is set to the value @production@, or is unset, then the @zone@ is @production@. Otherwise, the value of @zone@ is the value of the @ClientOptions#zone@ option.
+**** @(RAS2b1a)@ A @zone@ value of @production@ corresponds to a PD of @ably.io@.
+**** @(RAS2b1b)@ A @zone@ value of @dev@ corresponds to a PD of @ably-dev.net@.
```

Comment:
Is it an unneccessary complication in client libraries to have them understand `zone` as a concept? AIUI all the client library is doing with it is mapping it to a PD and a fallbackPD. Might it  be simpler to just have clientOptions for `primaryDomain`  and `fallbackDomain`, and assume the ones for production if those aren't specified? Given that the only users of the `dev` zone will be us. (Keep RAS2b2c for calculating the host when a custom domain is specified).